apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -1245,6 +1251,58 @@ impl Telemetry {
         };
         sender.send(SingleReport::Stats(metrics));
     }
+
+    fn decode_subgraph_ftv1(context: &Context) -> (u64, HashMap<String, SingleTypeStat>) {
+        use crate::plugins::telemetry::apollo_exporter::proto::reports::trace::Node;
+
+        fn traverse(
+            per_type: &mut HashMap<String, SingleTypeStat>,
+            field_execution_weight: f64,
+            node: &Node,
+        ) {
+            let single_field_stat = per_type
```

Comment:
Not every node in this tree represents a field/resolver: there's also nodes that represent list elements (and the root). So unless parent_type, field name (but see my next comment too) and the start/end times are set, you shouldn't update `per_type` at all, like in the AS code.